crates/cojson-core/src/core/cache.rs

```diff
@@ -0,0 +1,70 @@
+use ed25519_dalek::SigningKey;
+use lru::LruCache;
+use salsa20::{cipher::Key, XSalsa20};
+use std::{cell::RefCell, num::NonZero};
+
+use crate::core::{CoJsonCoreError, KeySecret, SignerSecret};
+
+#[derive(Debug, Clone)]
+pub struct CryptoCache {
+    xsalsa20_key_cache: RefCell<LruCache<KeySecret, Key<XSalsa20>>>,
+    ed25519_signing_key_cache: RefCell<LruCache<SignerSecret, SigningKey>>,
+}
+
+impl CryptoCache {
+    pub fn new() -> Self {
+        Self {
+            xsalsa20_key_cache: RefCell::new(LruCache::new(NonZero::new(2).unwrap())),
+            ed25519_signing_key_cache: RefCell::new(LruCache::new(NonZero::new(2).unwrap())),
+        }
+    }
+
+    /// Get or derive the XSalsa20 key from a KeySecret, using the cache.
+    pub fn get_xsalsa20_key(&self, key_secret: &KeySecret) -> Result<Key<XSalsa20>, CoJsonCoreError> {
+        let mut cache = self.xsalsa20_key_cache.borrow_mut();
```

Comment:
Please may you get rid of this cache entirely? If you look at the implementations of the `get_` methods it's basically just some byte copies at worst, which are likely to be optimized away anyways.

Reply:
Update: this cache does make sense because it sidesteps base58 encoding!